spotify_playlist/fetch_songs.py

```diff
@@ -0,0 +1,18 @@
+from c3po.db.base import session_scope
+from c3po.db.models.user import UserPosts
+from c3po.db.models.link import Link
+from spotify_uri import get_spotify_id
+
+# get first 50 songs from the db
+def get_song_url():
+    with session_scope() as session:
+        songs = []
+        for u,l in session.query(UserPosts,Link).filter(UserPosts.link_id==Link.id).limit(50).all():
```

Comment:
You are not using the `u` variable anywhere. It would be better if you queried the Songs table (which has entries that definitely have corresponding Spotify IDs). Do a join with the Links table and then get the URLs from there. This would ensure you get 50 songs in a single go.